### PR TITLE
Assignment2 - Ethan Florko

### PR DESCRIPTION
**Pull request by EthanFlorko**

### CODE REVIEW

**Comment by Aramirez61** [comment]
I'll review this.

File: Logger.Tests/Logger.Tests.csproj [hunk 1]
```diff
@@ -4,6 +4,17 @@
 		<TargetFramework>net7.0</TargetFramework>
```
Comment (Aramirez61) [inline Logger.Tests/Logger.Tests.csproj:4]: May want to change target framework to net 8.0 and lang version to 12.0.

File: Logger.Tests/FileLoggerTests.cs [hunk 2]
```diff
@@ -1,9 +1,38 @@
...
-public class FileLoggerTests
+namespace Logger.Tests
 {
 
-}
+    [TestClass]
+    public class FileLoggerTests
+    {
+        [TestMethod]
+        public void Log_ValidInput_LogsMessageToFile()
```
Comment (Aramirez61) [inline Logger.Tests/FileLoggerTests.cs:13]: Good use of following the unit test naming conventions.

File: Logger.Tests/LogFactoryTests.cs [hunk 3]
```diff
@@ -5,5 +5,32 @@ namespace Logger.Tests;
...
+        //Act
+        var logger = logFactory.CreateLogger("ExampleClass");
+        
+        //Assert
+        Assert.IsNotNull(logger);
+        Assert.IsInstanceOfType(logger, typeof(FileLogger));
+    }
+    
+    [TestMethod]
+    public void CreateLogger_WithoutConfiguredFileLogger_ReturnsNull()
```
Comment (Aramirez61) [inline Logger.Tests/LogFactoryTests.cs:25]: Good testing of covering cases when ConfigureFileLogger is not used.

File: Logger/BaseLoggerMixins.cs [hunk 4]
```diff
@@ -1,6 +1,49 @@
...
-public static class BaseLoggerMixins
+namespace Logger
 {
 
-}
+    public static class BaseLoggerMixins
+    {
+        public static void Error(this BaseLogger logger, string message, params object[] args)
+        {
+            LogWithLevel(logger, LogLevel.Error, message, args);
```
Comment (Aramirez61) [inline Logger/BaseLoggerMixins.cs:13]: I think all of these LogWithLevel calls should be replaced with baseLogger.Log() method calls. I am not able to see where the LogWithLevel method fits into BaseLoggerMixins class.

**Review by Aramirez61 — COMMENTED** [review]
There is an existing BaseLogger class. It needs an auto property to hold a class name. This property should be set in the LogFactory using an object initializer. ✔

Create a FileLogger that derives from BaseLogger. It should take in a path to a file to write the log message to. When its Log method is called, it should append messages on their own line in the file. The output should include all of the following:
The current date/time ✔
The name of the class that created the logger ✔
The log level ✔
The message ✔
The format may vary, but an example might look like this "10/7/2019 12:38:59 AM FileLoggerTests Warning: Test message"
The LogFactory should be updated with a new method ConfigureFileLogger. This should take in a file path and store it in a private member. It should use this when instantiating a new FileLogger in its CreateLogger method. ✔
If the file logger has not be configured in the LogFactory, its CreateLogger method should return null. ✔
Inside of BaseLoggerMixins implement extension methods on BaseLogger for
Error, ❌
Warning, ❌
Information, and ❌
Debug. ❌ Each of these methods should take in a string for the message, as well as a parameter array of arguments for the message. Each of these extension methods is expected to be a shortcut for calling the BaseLogger.Log method, by automatically supplying the appropriate LogLevel. These methods should throw an exception if the BaseLogger parameter is null. There are a couple example unit tests to get you started.
-
(NOTE: extension methods are implemented here but need to be using the BaseLogger.log, which is not currently being used in this BaseLoggerMixins class).

Use the nameof() operator when identifying the class name to the logger ✔
Ensure you turn on Warnings as Errors (TreatWarningsAsErrors) ✔
Ensure that you turn on code analysis (EnableNETAnalyzers) ✔
Ensure that you turn on CodeAnalysisTreatWarningsAsErrors ✔
Ensure that you turn on EnforceCodeStyleInBuild ✔
Set LangVersion and the TargetFramework to the latest released versions available (preview versions optional) ❌
Turn on Nullability (Nullable) ✔
All of the above should be unit tested.
Extra Credit
Implement an additional logger. This logger must be unit tested. Some options to consider could be one that uses System.Console or System.Diagnostics.Trace ❌
Implement the factory pattern using static abstract methods on a Logger interface instead.

File: Logger/FileLogger.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+{
+    public class FileLogger : BaseLogger
+    {
+        //Private mem to store file path
+        private readonly string filePath;
+
+        //Constructor
+        public FileLogger(string filePath)
+        {
+            this.filePath = filePath;
```
Comment (BenjaminMichaelis) [inline Logger/FileLogger.cs:14]: `this` is not needed if the field is named properly with a leading underscore

**Pull request closed without merging**